Feature / Widget redesign

benjay10 opened the pull request:
This plugin now features two extra buttons in the new "Insert" menu to insert a date and a dateTime value. The contextual card appears as normal to further insert a date, but with some bugfixes for the stuck `onlyDate` state. Also proper translations are provided and the plugin uses more Ember Appuniversum components.

Commit:
Update dependencies

An new version of the editor is out, and things need to be tested.

Commit:
Clean up unneeded styling

Commit:
Fix this.marks undefined in insertText operation

Not sure if this is perfect solution. Don't even know for sure what a
mark is, but looking at other code, this seems to have been used before.

Commit:
Some linting fixes

Commit:
Compatible version of package-lock file

Accidentally committed wrong format of the package-lock file in previous
commits.

Commit:
Remove ember-dependency-lint

This linting test is not part of the normal Ember addon temp and is
blocking builds. I think it would be better to do this linting on the
top level project (editor, or frontend-gn) rather than only on this
specific plugin, or maybe on every plugin at once. It's just too
inconsistent for now.

Commit:
Added new component with translations

Commit:
Added more translations

Commit:
Fixed problem with 2-way binding

Commit:
Fixed tracking issue on time picker

Commit:
New time changed callback, but still broken

Some more testing and potentially updates to Ember Appuniversum needed
to fix this.

Commit:
Some redesigning the components

Redesign using Appuniversum form rows, and datepicker components.

Commit:
Local tracking and cleanup

With the local tracking, one can update the arguments from outside the
component (e.g. when a new date in the editor is selected and the card
needs to show updated values) but also keep local state (for when you
are still selecting the correct date).

Commit:
Cleanup and only show insert button when needed

Only show the insert button when there is not already a date in the
documnet, in which case we just want to updated the dat directly.

Commit:
Included tracked-toolbox

benjay10 commented:
**Blocked by:**
In order for this plugin to work correctly, we need to wait for a review of the AuTimepicker component. Progress is potentially on its way here: https://github.com/appuniversum/ember-appuniversum/pull/252.
Also need to wait for a new release and include that in the this PR.

Dietr commented:
For these compact cards I prefer not showing the card header and removing the expandable option. Makes it's function easier to discover and the card more compact. What do you think?

For example:
![Screenshot 2022-05-13 at 14 54 41](https://user-images.githubusercontent.com/2385322/168287574-98ae29a9-3c95-48c8-a5ef-789ce526e777.png)

Dietr commented:
There seems to be an overflow issue with the .say-container__aside div. Not to be fixed here (will add a fix on ember-rdfa-editor). It cuts off the datepicker.

![Screenshot 2022-05-13 at 14 53 01](https://user-images.githubusercontent.com/2385322/168287870-f6cade25-eb02-415f-be69-d4c832d27e71.png)

Dietr commented on translations/nl_BE.yaml:
```diff
@@ -0,0 +1,14 @@
+    insert-datetime: "Datum en tijd invoegen"
+    label: "Selecteer datum"
+    hours: "uur"
+    minutes: "minuten"
```
Maybe we should use a capital for "Uur" and "Minuten"? Labels always start with capitals in our apps.

benjay10 commented:
@Dietr Indeed the datepicker being cut of is annoying, and I will also update the translations to have capitalisation.

For the collapsing: some people would possibly prefer less clutter on screen and would like the collapsing cards, although, without the collapsing, it is sleeker, more discoverable and you need less clicks to interact with dates. I can agree with both side and so I would choose for the design that fits in best with the rest of the editor and the desing language it is going for in the future.

Commit:
Capitalisation in translation, new dependencies

This version of the date plugin should depend on a later version of the
ember-appuniversum framwork to benefit from the corrected AuTimepicker.

benjay10 commented:
@Dietr If you want, you can commit the new lay-out you demonstrated in the image above if you have those changes.

Dietr committed:
update ember-appuniversum, hide card header
